File: 01/004/main.r
isPalindrome <- function(num) {
    str <- toString(num)
    str == paste(rev(strsplit(str, "")[[1]]), collapse="")
}

largestPalindrome <- function(num) {
    result <- 0
    for (i in num:0) {
        if (isPalindrome(num * i)) {
            result <- num * i
            break
        }
    }
    ifelse(num==0, 0, max(result, largestPalindrome(num-1)))
}

print(largestPalindrome(999))

File: 01/004/main.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func IsPalindrome(num int) bool {
	var str = strconv.Itoa(num)
	for i := 0; i < len(str)/2; i++ {
		if str[i] != str[len(str)-i-1] {
			return false
		}
	}
	return true
}

func LargestPalindrome(num int) int {
	if num == 0 {
		return 0
	}
	var result = 0
	for i := num; i > 0; i-- {
		if IsPalindrome(num * i) {
			result = num * i
			break
		}
	}
	return int(math.Max(float64(result), float64(LargestPalindrome(num-1))))
}

func main() {
	fmt.Println(LargestPalindrome(999))
}
